### PR TITLE
Fix test order in /pkg/probe/http/http_test.go

### PR DESCRIPTION
**Pull request by lunhuijie**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
Per https://pkg.go.dev/github.com/stretchr/testify/assert#Equal expected goes before actual:

`func Equal(t TestingT, expected, actual interface{}, msgAndArgs ...interface{}) bool`
#### Which issue(s) this PR fixes:

Fixes #

#### Special notes for your reviewer:
Fixes tests asserting that the expected values were the actual ones, when they were not. just like https://github.com/kubernetes/kubernetes/pull/102611
#### Does this PR introduce a user-facing change?

```release-note
"NONE"
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs
```

### CODE REVIEW

File: pkg/probe/http/http_test.go [hunk 1]
```diff
@@ -508,8 +508,8 @@ func TestHTTPProbeChecker_PayloadTruncated(t *testing.T) {
 		require.NoError(t, err)
 		result, body, err := prober.Probe(target, headers, wait.ForeverTestTimeout)
 		assert.NoError(t, err)
-		assert.Equal(t, result, probe.Success)
-		assert.Equal(t, body, string(truncatedPayload))
+		assert.Equal(t, probe.Success, result)
+		assert.Equal(t, string(truncatedPayload), body)
```
Comment (yangjunmyfm192085) [inline pkg/probe/http/http_test.go:512]: It seems good

**Review by ehashman — COMMENTED** [review]
/priority backlog
/triage accepted
/lgtm

**Review by mrunalp — APPROVED** [review]

**Pull request merged**